packages/HorizonUI/Sources/HorizonUI/Sources/Components/ProgressBar/HorizonUI.ProgressBar.swift

```diff
@@ -0,0 +1,125 @@
+
+import SwiftUI
+
+public extension HorizonUI {
+    struct ProgressBar: View {
+        // MARK: - Dependancies
+
+        private let progress: Double
+        private let size: Size
+        private let progressColor: Color
+        private let numberPosition: NumberPosition
+        private let textColor: Color
+
+        // MARK: - Init
+
+        /// - Parameters:
+        ///    - progress : The percentage value, must be from 0 to 1.
+        ///    - progressColor : The color for progress bar
+        ///    - size: Select from two values [small or medium]
+        ///    - numberPosition: The progress text position select from three values [inside, outside or hidden]
+        ///    - textColor: The color for the progress bar text
+        ///
+        /// - Example:
+        /// HorizonUI.ProgressBar(
+        ///     progress: 0.5,
+        ///     progressColor:  .huiColors.surface.institution,
+        ///     size: .medium,
+        ///     numberPosition: .outside,
+        ///     textColor: .huiColors.primitives.white10
+        ///  )
+        public init(
+            progress: Double,
+            progressColor: Color = .huiColors.surface.institution,
+            size: Size,
+            numberPosition: NumberPosition = .inside,
+            textColor: Color = .huiColors.surface.institution
+        ) {
+            self.progress = progress
+            self.size = size
+            self.progressColor = progressColor
+            self.numberPosition = numberPosition
+            self.textColor = textColor
+        }
+
+        public var body: some View {
+            HStack(spacing: .huiSpaces.primitives.xSmall) {
+                GeometryReader { geometry in
+                    progressContentView(width: geometry.size.width * progress)
+                }
+
+                if numberPosition == .outside {
+                    progressText
+                }
+            }
+            .frame(height: size.height)
+        }
+    }
+}
+
+// MARK: - Components
+
+extension HorizonUI.ProgressBar {
+    private func progressContentView(width: CGFloat) -> some View {
+        backgroundView
+            .overlay(alignment: .leading) {
+                ZStack(alignment: .leading) {
+                    progressFillView(width: width)
+
+                    if numberPosition == .inside {
+                        progressText
+                            .frame(minWidth: width, alignment: .trailing)
+                    }
+                }
+                .huiCornerRadius(level: .level28)
+            }
+    }
+
+    @ViewBuilder
+    private var progressText: some View {
+        if numberPosition != .hidden {
+            let percentageRound = round(progress * 10000) / 100.0
+            Group {
+                Text(percentageRound, format: .number) + Text("%")
+            }
+            .padding(.horizontal, .huiSpaces.primitives.xSmall)
+            .foregroundStyle(textColor)
+            .font(.huiFonts.figtreeSemibolt14)
```

Comment:
Please use the `buttonTextMedium` typography modifier on text elements instead of setting the font.